src/Files/Filesystem/FolderSizeProvider.cs

```diff
@@ -174,5 +174,10 @@ private void PreferencesSettingsService_PropertyChanged(object sender, PropertyC
                 RaiseSizeChanged(null);
             }
         }
+
+        public void Dispose()
+        {
+            preferencesSettingsService.PropertyChanged -= PreferencesSettingsService_PropertyChanged;
```

Comment:
This is mostly unnecessary because there is only one global instance of FolderSizeProvider which gets never Diposed. But surely doesn't hurt.